src/test/java/com/jadajabajava/deserializers/HhVacancyDeserializerTest.java

```diff
@@ -0,0 +1,67 @@
+package com.jadajabajava.deserializers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.jadajabajava.entities.Vacancy;
+import com.jadajabajava.services.EmployerService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ExtendWith(MockitoExtension.class)
+class HhVacancyDeserializerTest {
+
+    @Mock
+    private EmployerService employerService;
+
+    @InjectMocks
+    private HhVacancyDeserializer vacancyDeserializer;
+
+    private static final String VACANCY_VALID_JSON_FILE_PATH = "json/vacancy_valid.json";
+    private static final String VACANCY_INVALID_JSON_FILE_PATH = "json/vacancy_invalid.json";
+
+    private final ClassLoader classLoader = this.getClass().getClassLoader();
+
+    private ObjectMapper mapper;
+
+    @BeforeEach
+    void supplyDeserializer() {
+        SimpleModule simpleModule = new SimpleModule();
+        simpleModule.addDeserializer(Vacancy.class, vacancyDeserializer);
+        mapper = new ObjectMapper();
+        mapper.registerModule(simpleModule);
+    }
+
+    @Test
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5720132)